test/e2e/logstash/stack_monitoring_test.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build logstash || e2e
+
+package logstash
+
+import (
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/stackmon/validations"
+	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test"
+	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test/checks"
+	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test/elasticsearch"
+	"github.com/elastic/cloud-on-k8s/v2/test/e2e/test/logstash"
+	"testing"
+)
+
+// TestESStackMonitoring tests that when an Logstash is configured with monitoring, its log and metrics are
+// correctly delivered to the referenced monitoring Elasticsearch clusters.
+func TestLogstashStackMonitoring(t *testing.T) {
+	// only execute this test on supported version
+	err := validations.IsSupportedVersion(test.Ctx().ElasticStackVersion)
+	if err != nil {
+		t.SkipNow()
+	}
+
+	// create 1 monitored and 2 monitoring clusters to collect separately metrics and logs
+	metrics := elasticsearch.NewBuilder("test-ls-mon-metrics").
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources)
+	logs := elasticsearch.NewBuilder("test-ls-mon-logs").
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources)
+	monitored := logstash.NewBuilder("test-ls-mon-a").
+		WithNodeCount(1).
+		WithMonitoring(metrics.Ref(), logs.Ref()).
+		//TODO: remove command when Logstash has built with a monitor version of log4j2.properties
```

Comment:
I'm in two minds as to whether we should include support for log monitoring, given the current lack of upstream support in the docker image to supply a working log4j2.properties file, leading to this requirement to manually add a log4j2.properties file. 

In this case, I get nervous of having this dependency on downloading an upstream logstash log4j2.properties file in order for the e2e tests to pass.

Reply:
From the operator side, we don't limit the feature if users have a custom docker image. Customers may already have one in production. I am open to remove log monitoring.

The upstream log4j2.properties is under Logstash team control. I have fewer security concerns here.

Reply:
My concern was more around having a test dependency on an artifact from the `main` branch of a separate codebase, which may result in unintended, and difficult to track down, test failures in this codebase if we ever change the log4j2.properties file in that branch. 

What is the behaviour if the `log:` monitoring stanza is created without the log4j2.properties addition to the container? `filebeat` sidecar gets created but just doesn't do anything?

Reply:
Without the file, Logstash won't write log to file, so `filebeat` sideccar is created but does nothing.

Would it help if the download change to `v8.6.1` instead of `main`? The file won't change and the test result should be consistent.

Reply:
I think that's fine for now. I'd like to create a couple of issues to track this - on the logstash side, an issue to track the update to the docker container to support an environment variable to use the file-based log4j2.properties, and a linked `TODO` to to track removing it in this codebase